internal/platform/storage/couchbase.go

```diff
@@ -0,0 +1,258 @@
+
+	_, err = users.Upsert(id.String(), user, 0)
+	if err != nil {
+		return errors.Wrap(err, "failed to upsert user")
+	}
+
+	return nil
+}
+
+func (s *Storage) GetUserByID(id string) (User, error) {
+	var user User
+
+	query := gocb.NewN1qlQuery("select Email from users where ID=$userID")
+
+	params := make(map[string]interface{})
+	params["userID"] = id
+
+	users, err := s.cluster.OpenBucket("users", "")
+	if err != nil {
+		return user, errors.Wrap(err, "failed to open bucket")
+	}
+
+	defer users.Close()
+
+	rows, err := users.ExecuteN1qlQuery(query, params)
+	if err != nil {
+		return user, errors.Wrap(err, "can't get user")
+	}
+
+	rows.Close()
+
+	if rows.Metrics().ResultCount == 0 {
+		return user, ErrUserNotFound
+	}
+
+	for rows.Next(&user) {
+		break
+	}
+
+	return user, nil
+}
+
+func (s *Storage) GetUserByEmail(email string) (User, error) {
+	var user User
+
+	query := gocb.NewN1qlQuery("select `ID`,`Email`,`Password` from `users` where `Email`=$email")
+
+	params := make(map[string]interface{})
+	params["email"] = email
+
+	users, err := s.cluster.OpenBucket(usersBucketName, "")
+	if err != nil {
+		return user, errors.Wrap(err, "can't open 'users' bucket")
+	}
+
+	defer users.Close()
+
+	rows, err := users.ExecuteN1qlQuery(query, params)
+	if err != nil {
+		return user, errors.Wrap(err, "can't get user")
+	}
+
+	rows.Close()
+
+	if rows.Metrics().ResultCount == 0 {
+		return User{}, ErrUserNotFound
+	}
+
+	for rows.Next(&user) {
+		break
+	}
+
+	return user, nil
+}
+
+func (s *Storage) CreateToken(token string, claims Claims) error {
+	tokenData := Token{
+		Token:  token,
+		Claims: claims,
+	}
+
+	tokens, err := s.cluster.OpenBucket(tokensBucketName, "")
+	if err != nil {
+		return errors.Wrap(err, "can't open 'tokens' bucket")
+	}
+
+	defer tokens.Close()
+
+	_, err = tokens.Upsert(token, tokenData, 0)
+	if err != nil {
+		return errors.Wrap(err, "can't upset token")
+	}
+
+	return nil
+}
+
+func (s *Storage) DeprecateToken(token Token) error {
+	// set token expiration time if far far past
+	token.Claims.ExpiresAt = 1
+
+	tokens, err := s.cluster.OpenBucket(tokensBucketName, "")
+	if err != nil {
+		return errors.Wrap(err, "can't open 'tokens' bucket")
+	}
+
+	defer tokens.Close()
+
+	_, err = tokens.Upsert(token.Token, token, 0)
+	if err != nil {
+		return errors.Wrap(err, "can't upsert token")
+	}
+
+	return nil
+}
+
+func (s *Storage) GetNotExpiredTokenByToken(token string) (Token, error) {
+	query := gocb.NewN1qlQuery("select claims, token from tokens where claims.exp > $now and token = $token")
+
+	params := make(map[string]interface{})
+	params["token"] = token
+	params["now"] = time.Now().Unix()
+
+	tokens, err := s.cluster.OpenBucket(tokensBucketName, "")
+	if err != nil {
+		return Token{}, errors.Wrap(err, "can't open 'tokens' bucket")
+	}
+
+	defer tokens.Close()
+
+	rows, err := tokens.ExecuteN1qlQuery(query, params)
+	if err != nil {
+		return Token{}, errors.Wrap(err, "can't get token")
+	}
+
+	rows.Close()
```

Comment:
probably you forgot defer

Reply:
No I don't. I need to execute this method as it is now.

Reply:
investigation needed #7